flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/NettyMessageClientDecoderDelegateTest.java

```diff
@@ -0,0 +1,369 @@
+
+		testRepartitionMessagesAndDecode(
+			channel,
+			true,
+			false,
+			false,
+			normalInputChannel.getInputChannelId(),
+			null);
+	}
+
+	/**
+	 * Verifies that NettyMessageDecoder works well with buffers sent to a released input channel. The data buffer
+	 * part should be discarded before reading the next message.
+	 */
+	@Test
+	public void testDownstreamMessageDecodeWithReleasedInputChannel() throws Exception {
+		int totalBufferRequired = 3;
+
+		SingleInputGate normalInputGate = createSingleInputGate(1);
+		RemoteInputChannel normalInputChannel = new BufferProviderRemoteInputChannel(normalInputGate, totalBufferRequired, BUFFER_SIZE);
+
+		SingleInputGate releasedInputGate = createSingleInputGate(1);
+		RemoteInputChannel releasedInputChannel = new BufferProviderRemoteInputChannel(releasedInputGate, 0, BUFFER_SIZE);
+		releasedInputGate.close();
+
+		CreditBasedPartitionRequestClientHandler handler = new CreditBasedPartitionRequestClientHandler();
+		handler.addInputChannel(normalInputChannel);
+		handler.addInputChannel(releasedInputChannel);
+
+		EmbeddedChannel channel = new EmbeddedChannel(new NettyMessageClientDecoderDelegate(handler));
+
+		testRepartitionMessagesAndDecode(
+			channel,
+			false,
+			true,
+			false,
+			normalInputChannel.getInputChannelId(),
+			releasedInputChannel.getInputChannelId());
+	}
+
+	/**
+	 * Verifies that NettyMessageDecoder works well with buffers sent to a removed input channel. The data buffer
+	 * part should be discarded before reading the next message.
+	 */
+	@Test
+	public void testDownstreamMessageDecodeWithRemovedInputChannel() throws Exception {
+		int totalBufferRequired = 3;
+
+		SingleInputGate normalInputGate = createSingleInputGate(1);
+		RemoteInputChannel normalInputChannel = new BufferProviderRemoteInputChannel(normalInputGate, totalBufferRequired, BUFFER_SIZE);
+
+		CreditBasedPartitionRequestClientHandler handler = new CreditBasedPartitionRequestClientHandler();
+		handler.addInputChannel(normalInputChannel);
+
+		EmbeddedChannel channel = new EmbeddedChannel(new NettyMessageClientDecoderDelegate(handler));
+
+		testRepartitionMessagesAndDecode(
+			channel,
+			false,
+			false,
+			true,
+			normalInputChannel.getInputChannelId(),
+			null);
+	}
+
+	//------------------------------------------------------------------------------------------------------------------
+
+	private void testRepartitionMessagesAndDecode(
+		EmbeddedChannel channel,
+		boolean hasEmptyBuffer,
+		boolean hasBufferForReleasedChannel,
+		boolean hasBufferForRemovedChannel,
+		InputChannelID normalInputChannelId,
+		InputChannelID releasedInputChannelId) throws Exception {
+
+		List<NettyMessage> messages = createMessageList(
+			hasEmptyBuffer,
+			hasBufferForReleasedChannel,
+			hasBufferForRemovedChannel,
+			normalInputChannelId,
+			releasedInputChannelId);
+
+		ByteBuf[] serializedBuffers = null;
+		ByteBuf mergedBuffer = null;
+
+		try {
+			serializedBuffers = serializeMessages(messages);
```

Comment:
serializeMessages -> encodeMessages to consistent with below `decodedMessages`

Reply:
Renamed the method and related variables, also rename `decodedMessages` to `decodeMessages`